src/x/net/http/convert.go

```diff
@@ -38,6 +39,77 @@ var (
 	errDurationType = errors.New("invalid duration type")
 )
 
+// NanosToDurationBytes transforms a json byte slice with Nano keys into Duration keys.
+func NanosToDurationBytes(r io.Reader) (map[string]interface{}, error) {
+	var dict map[string]interface{}
+	d := json.NewDecoder(r)
+	d.UseNumber()
+	if err := d.Decode(&dict); err != nil {
+		return nil, fmt.Errorf("error decoding JSON: %s", err.Error())
+	}
+
+	ret, err := NanosToDurationMap(dict)
+	if err != nil {
+		return nil, err
+	}
+
+	return ret, nil
+}
+
+// NanosToDurationMap transforms keys with Nanos into Duration.
+func NanosToDurationMap(input map[string]interface{}) (map[string]interface{}, error) {
```

Comment:
Do you need to preserve `input`? If these payloads are large, it might be better to do `ConverToDurationMap` or similar and just add/delete as necessary instead of using 2x the memory.

Reply:
Nah these are super small. This endpoint just lists out the namespaces in m3db.